scripts/registry-validate-source.py

```
@@ -55,12 +57,25 @@ def _expect_optional_str_array(obj: dict, key: str, ctx: str) -> None:
         return
     value = obj[key]
     if not isinstance(value, list) or not all(isinstance(item, str) and item for item in value):
-        raise ValidationError(f"{ctx}.{key} must be a string array")
+        raise ValidationError(f"{ctx}.{key} must be a non-empty string array")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)